### PR TITLE
[WIP] Fix update steps for SA

### LINKED ISSUE #98415: SA has multiple secrets
**What happened**:

``` yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"ServiceAccount","metadata":{"annotations":{},"name":"test","namespace":"default"}}
  creationTimestamp: "2021-01-26T09:39:09Z"
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:kubectl.kubernetes.io/last-applied-configuration: {}
    manager: kubectl-client-side-apply
    operation: Update
    time: "2021-01-26T09:39:09Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:secrets:
        .: {}
        k:{"name":"test-token-76ggg"}:
          .: {}
          f:name: {}
        k:{"name":"test-token-97zg5"}:
          .: {}
          f:name: {}
        k:{"name":"test-token-g9ndg"}:
          .: {}
          f:name: {}
        k:{"name":"test-token-g9zmx"}:
          .: {}
          f:name: {}
        k:{"name":"test-token-mxx7p"}:
          .: {}
          f:name: {}
        k:{"name":"test-token-mzxs8"}:
          .: {}
          f:name: {}
    manager: kube-controller-manager
    operation: Update
    time: "2021-01-26T11:11:49Z"
  name: test
  namespace: default
  resourceVersion: "1993939"
  uid: 8aad69fa-8b3d-48c2-adc6-af6a50448127
secrets:
- name: test-token-g9ndg
- name: test-token-mxx7p
- name: test-token-g9zmx
- name: test-token-97zg5
- name: test-token-mzxs8
- name: test-token-76ggg
```

**What you expected to happen**:

```shell
while true ; do
kubectl delete secrets --all
done
```

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.20.1
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`): 7
- Kernel (e.g. `uname -a`):  Linux zsm-test-2 3.10.0-1160.11.1.el7.x86_64 SMP Fri Dec 18 16:34:56 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by wzshiming**
**What type of PR is this?**
/kind bug
/sig api-machinery

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #98415

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Fix update steps for SA
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: pkg/controller/serviceaccount/tokens_controller.go [hunk 1]
```diff
@@ -707,6 +707,29 @@ func (e *TokensController) listTokenSecrets(serviceAccount *v1.ServiceAccount) (
 	return items, nil
 }
 
+func (e *TokensController) updateServiceAccount(ctx context.Context, namespace string, sa *v1.ServiceAccount) error {
+	// Remove the secret if not exist.
+	if len(sa.Secrets) != 0 {
+		var refs []v1.ObjectReference
+		for _, ref := range sa.Secrets {
+			_, err := e.client.CoreV1().Secrets(namespace).Get(ctx, ref.Name, metav1.GetOptions{})
```
Comment (liggitt) [inline pkg/controller/serviceaccount/tokens_controller.go:715]: doing N live gets on every service account update is a significant change… do you have details about the bug this is trying to fix? A test demonstrating that bug would be a helpful starting point

Comment (wzshiming) [reply pkg/controller/serviceaccount/tokens_controller.go:715]: #98415 it to reproduce this bug 
``` shell
while true ; do
kubectl delete secrets --all
done
```

I triggered this problem while upgrading k8s and needed to recreate the SA to fix it

**Pull request closed without merging**